src/ape/api/transactions.py

```diff
@@ -200,6 +203,56 @@ def decode_logs(self, abi: Union[EventABI, "ContractEvent"]) -> Iterator[Contrac
 
         yield from self.provider.network.ecosystem.decode_logs(abi, self.logs)
 
+    @property
+    def events(self):
+        contracts = {log["address"] for log in self.logs}
+        contract_types = {addr: self.chain_manager.contracts.get(addr) for addr in contracts}
+        decoded = []
+        for log in self.logs:
+            note = self.decode_ds_note(contract_types[log["address"]], log)
+            if note:
+                decoded.append(note)
+            else:
+                try:
+                    event_abi = next(
```

Comment:
This could be simplified if we add choice by selector to ethpm_types. I see it like this:
```python
ContractType.events["Transfer"]
ContractType.events[selector]
ContractType.events[keccak(selector)]
```
I'll make a pull request.